Userland/Utilities/touch.cpp

```diff
@@ -1,51 +1,253 @@
 /*
  * Copyright (c) 2018-2021, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2022, Ariel Don <ariel@arieldon.com>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#include <AK/CheckedFormatString.h>
+#include <AK/GenericLexer.h>
+#include <AK/Time.h>
 #include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
 #include <LibCore/System.h>
 #include <LibMain/Main.h>
+#include <LibTimeZone/TimeZone.h>
+#include <ctype.h>
```

Comment:
Thas seems unnecessary

Reply:
`parse_datetime()` uses `isdigit()` from `ctype.h` in two cases.

1. `lexer.next_is(isdigit)`
2. `auto fractional_second = lexer.consume_while(isdigit);`

In the first case, I can replace `isdigit` with a string "01234789" and maintain the same functionality since we overload `next_is()` with a few different types. In the second case, `consume_while()` demands a callable. I can write a local lambda function instead that matches `isdigit()`.

```c++
auto is_digit = [](char c) { return c >= '0' && c <= '9'; }
```
I can also overload `consume_while()` in `AK/GenericLexer.h`.

```c++
StringView consume_while(char const* accepted)
{
    size_t start = m_index;
    while (!is_eof() && next_is(accepted))
        ++m_index;
    size_t length = m_index - start;

    if (length == 0)
        return {};
    return m_input.substring_view(start, length);
}
```

This way, I can replace `isdigit` with "01234789" in both cases. It doesn't seem like the intended behavior of `GenericLexer` though since none of the similar functions (`consume_until()`, `ignore_while()`, `ignore_until()`) have this overload. Plus, each would need a similar StringView overload as well for symmetry -- that's a lot of added code left unused.

I can also pass the local lambda `is_digit()` to `next_is()`.

Either works for me: local lambda or new overloads -- your pick. I can also leave it as is with `isdigit()`.

Reply:
I guess then it's fine.

Reply:
It's good that you mentioned this regardless because I found a separate bug while messing with this. Previously, when parsing the fractional second specified, an empty time zone was not permitted, but it should be -- I updated this.

I pushed that change. Is there anything else to update before merging?